dataflux_core/download.py

```diff
@@ -156,6 +160,50 @@ def __init__(self, max_composite_object_size):
         self.max_composite_object_size = max_composite_object_size
 
 
+def dataflux_download_parallel(
+    project_name: str,
+    bucket_name: str,
+    objects: list[tuple[str, int]],
+    storage_client: object = None,
+    dataflux_download_optimization_params: DataFluxDownloadOptimizationParams = None,
+    parallelization: int = 1,
```

Comment:
Can we create follow-up tickets to document the work needed for this parameter to be automatically and intelligently decided by the `dataflux-python-client`? Since we are recommending customers to modify the source code of the Dataflux dataset, it would be hard for them to understand when and how to use this parameter.

Ideally, the dataflux download function should be intelligent enough to tell whether to create additional processes or not, based on the combination of factors such as available memory (don't want to OOM for additional overhead), batch size (as we discussed, we don't want to create further processes if we are only to download 64 objects), etc.

Reply:
Yes follow up ticket is a great idea. My plan moving forward is to create a smart wrapper function that can take in a parameters like "batch size" and "ram" to figure out optimal parallelization. We can then provide that call to the dataflux pytorch library directly. That's why i'm putting this in a separate method so its only exposed to people working directly with the client.